docs/src/main/tut/TypedML.md

```diff
@@ -15,17 +29,226 @@ spark.sparkContext.setLogLevel("WARN")
+val indexerModel = indexer.fit(dataWithFeatures).run()
+
+case class IndexedDataWithFeatures(field1: Double, field2: Int, field3: String, features: Vector, indexedField3: Double)
+val indexedData = indexerModel.transform(dataWithFeatures).as[IndexedDataWithFeatures]
+```
+
+Then, we train the model:
+
+```tut:book
+case class RFInputs(indexedField3: Double, features: Vector)
+val rf = TypedRandomForestClassifier[RFInputs]
+
+val model = rf.fit(indexedData).run()
+```
+
+### Prediction
+
+We now want to predict `field3` for `testData` using the previously trained model. Please note that, like Spark ML API,
+`testData` has a default value for `field3` (empty String in our case) that will be ignored at prediction time. We reuse 
+our `vectorAssembler` to compute the feature vector of `testData` and our `indexerModel` to index `field3`.
+
+```tut:book
+val testData = TypedDataset.create(Seq(
+  Data(0D, 10, "")
+))
+val testDataWithFeatures = vectorAssembler.transform(testData).as[DataWithFeatures]
+val indexedTestData = indexerModel.transform(testDataWithFeatures).as[IndexedDataWithFeatures]
+
+case class PredictionInputs(features: Vector, indexedField3: Double)
+val testInput = indexedTestData.project[PredictionInputs]
+
+case class PredictionResultIndexed(
+  features: Vector,
+  indexedField3: Double,
+  rawPrediction: Vector,
+  probability: Vector,
+  predictedField3Indexed: Double
+)
+val predictionDs = model.transform(testInput).as[PredictionResultIndexed]
+```
+
+Then, we use a `TypedIndexToString` to get back a String value from `predictedField3`. `TypedIndexToString` takes
+as input the label array computed by our previous `indexerModel`:
+
+```tut:book
+case class IndexToStringInput(predictedField3Indexed: Double)
+val indexToString = TypedIndexToString[IndexToStringInput](indexerModel.transformer.labels)
+
+case class PredictionResult(
+  features: Vector,
+  indexedField3: Double,
+  rawPrediction: Vector,
+  probability: Vector,
+  predictedField3Indexed: Double,
+  predictedField3: String
+)
+val stringPredictionDs = indexToString.transform(predictionDs).as[PredictionResult]
+
+val prediction = stringPredictionDs.select(stringPredictionDs.col('predictedField3)).collect.run()
+
+prediction == Seq("foo")
+```
+
+## List of currently implemented `TypedEstimator`s
+
+* `TypedRandomForestClassifier`
+* `TypedRandomForestRegressor`
+* ... [your contribution here](https://github.com/typelevel/frameless/issues/215) ... :)
+
+## List of currently implemented `TypedTransformer`s
+
+* `TypedIndexToString`
+* `TypedStringIndexer`
+* `TypedVectorAssembler`
+* ... [your contribution here](https://github.com/typelevel/frameless/issues/215) ... :)
  
-## Using Vector and Matrix with TypedDataset
+## Using Vector and Matrix with `TypedDataset`
 
 `frameless-ml` provides `TypedEncoder` instances for `org.apache.spark.ml.linalg.Vector` 
 and `org.apache.spark.ml.linalg.Matrix`:
 
-```tut:book
+```tut:silent
```

Comment:
I think it's good to include here the imports a user needs to add. What do you think?

Reply:
`tut:silent` will indeed show the imports in the documentation (unlike `tut:invisible`), it's just that it will not add the interpreted lines `// import frameless.ml._` and `import org.apache.spark.ml.linalg._`.

Reply:
sorry, yes you are right